File: programmers/Level1/86491. 최소직사각형/86491. 최소직사각형.js
function solution(sizes) {
  let max_w = 0;
  let max_h = 0;
  sizes.forEach(([w, h]) => {
    if (w >= h) {
      max_w = Math.max(max_w, w);
      max_h = Math.max(max_h, h);
    } else {
      max_w = Math.max(max_w, h);
      max_h = Math.max(max_h, w);
    }
  });

  return max_w * max_h;
}

File: programmers/Level1/86491. 최소직사각형/86491. 최소직사각형.py
def solution(sizes):
    max_w, max_h = 0, 0
    
    for size in sizes:
        w, h = size
        
        if w >= h:
            max_w = max(max_w, w)
            max_h = max(max_h, h)
        
        else:
            max_w = max(max_w, h)
            max_h = max(max_h, w)
        
    answer = max_w * max_h

    return answer